File: my-folder/0034-find-first-and-last-position-of-element-in-sorted-array/solution.cpp
#include <vector>

class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int low = 0;
        int high = nums.size() - 1;
        int first = -1;
        int last = -1;
        while (low <= high) {
            int mid = low + (high - low) / 2;
            if (nums[mid] == target) {
                first = mid;
                last = mid;
                while (first > low && nums[first - 1] == target) {
                    first--;
                }
                while (last < high && nums[last + 1] == target) {
                    last++;
                }
                break; 
            } else if (nums[mid] > target) {
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }

        return {first, last};
    }
};

File: my-folder/0034-find-first-and-last-position-of-element-in-sorted-array/solution.java
class Solution {
    public int[] searchRange(int[] nums, int x) {
        int start =  -1;
        int end = -1;
        int left = 0,right=nums.length-1;
        while(left<=right){
            int mid = (left+right)/2;
            if(nums[mid]==x){
                start = mid;
                right = mid-1;
            }
            else if(nums[mid]<x) left = mid+1;
            else right = mid-1;
        }
        left = 0;
        right = nums.length-1;
        while(left<=right){
            int mid = (left+right)/2;
            if(nums[mid]==x){
                end = mid;
                left=mid+1;
            }
            else if(nums[mid]<x) left = mid+1;
            else right = mid-1;
        }
        return new int[]{start,end};
        
    }
}
